sys/kern/vm_map.c

```diff
@@ -220,28 +220,42 @@ void vm_map_delete(vm_map_t *map) {
   pool_free(P_VM_MAP, map);
 }
 
-/* TODO(fzdo): allow for changing protection bits of parts of entries */
-/* XXX: This function allows for setting protection bits fo existing entries
- * only. It can't change protection of part of entry (currently we don't need to
- * set protection of part of entry). */
 void vm_map_protect(vm_map_t *map, vaddr_t start, vaddr_t end, vm_prot_t prot) {
   SCOPED_MTX_LOCK(&map->mtx);
 
-#if 0
-  klog("vm_map_protect: 0x%x - 0x%x %c%c%c", start, end,
-       (prot & VM_PROT_READ) ? 'r' : '-', (prot & VM_PROT_WRITE) ? 'w' : '-',
-       (prot & VM_PROT_EXEC) ? 'x' : '-');
-#endif
+  /* Loop from first affected entry until we exit the affected region or next
+   * entry does not exist */
+  for (vm_map_entry_t *ent = vm_map_find_entry(map, start);
+       ent != NULL &&
+       (vm_map_entry_end(ent) > start && vm_map_entry_start(ent) < end);
```

Comment:
Can you make a helper function for checking that?

Reply:
Yes. Done.